[cloud_firestore]: 5.2.0 wont compile on windows

### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Which plugins are affected?

_No response_

### Which platforms are affected?

Windows

### Description

```
  firebase_core: ^3.3.0
  cloud_firestore: ^5.2.0
  firebase_storage: ^12.1.2
  firebase_auth: ^5.1.3
  firebase_analytics: ^11.2.1
  firebase_crashlytics: ^4.0.4
  cloud_functions: ^5.0.4
```

```
Launching lib\main.dart on Windows in debug mode...
Building Windows application...
Nuget is not installed.
Attempting to download nuget.
CMake Deprecation Warning at C:/Users/cyberpwn/Documents/development/workspace/Crucible/applications/crucible/build/windows/x64/extracted/firebase_cpp_sdk_windows/CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1576,40): error C2248: 'cloud_firestore_windows::PigeonDocumentSnapshot::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonDocumentSnapshot' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1585,24): error C2248: 'cloud_firestore_windows::PigeonDocumentChange::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonDocumentChange' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1591,60): error C2248: 'cloud_firestore_windows::PigeonSnapshotMetadata::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonSnapshotMetadata' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1683,22): error C2248: 'cloud_firestore_windows::PigeonDocumentSnapshot::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonDocumentSnapshot' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2293,1): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2293,56): error C2143: syntax error: missing ';' before '&' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2293,58): error C2065: 'request_arg': undeclared identifier [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2295,23): error C2146: syntax error: missing ';' before identifier 'encodable_request_arg' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2297,28): error C2065: 'request_arg': undeclared identifier [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
Error: Build process failed.
```

### Reproducing the issue

Remember google io years ago? Yeah do what they did. Try to actually use firestore on windows.

### Firebase Core version

3.3.0

### Flutter Version

3.22.3

### Relevant Log Output

```shell
Launching lib\main.dart on Windows in debug mode...
Building Windows application...
Nuget is not installed.
Attempting to download nuget.
CMake Deprecation Warning at C:/Users/cyberpwn/Documents/development/workspace/Crucible/applications/crucible/build/windows/x64/extracted/firebase_cpp_sdk_windows/CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1576,40): error C2248: 'cloud_firestore_windows::PigeonDocumentSnapshot::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonDocumentSnapshot' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1585,24): error C2248: 'cloud_firestore_windows::PigeonDocumentChange::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonDocumentChange' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1591,60): error C2248: 'cloud_firestore_windows::PigeonSnapshotMetadata::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonSnapshotMetadata' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\cloud_firestore_plugin.cpp(1683,22): error C2248: 'cloud_firestore_windows::PigeonDocumentSnapshot::ToEncodableList': cannot access private member declared in class 'cloud_firestore_windows::PigeonDocumentSnapshot' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2293,1): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2293,56): error C2143: syntax error: missing ';' before '&' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2293,58): error C2065: 'request_arg': undeclared identifier [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2295,23): error C2146: syntax error: missing ';' before identifier 'encodable_request_arg' [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2297,28): error C2065: 'request_arg': undeclared identifier [C:\Users\cyberpwn\Documents\development\workspace\Crucible\applications\crucible\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
Error: Build process failed.
```


### Flutter dependencies

This is a build error specifically about cloud firestore messages & the firestore plugin

### Additional context and comments

It's been years. Do you guys support firestore or firebase at all on windows or not? It seems like every version either fails to build or is extremely crashy on windows. It's a painful developer experience, and I feel like every time something like this is submitted its just kicked down the road until the automated bot closes the issue.

**Description:**

I'm encountering a build error when attempting to integrate the cloud_firestore plugin in my Flutter project on the Windows platform. The error occurs during the build process and appears to be related to the generated C++ code for the plugin.

**Environment:**

- Flutter Version: 3.4.3
- Dart SDK Version: 2.19.2
- cloud_firestore Version: 5.2.0 (also tried 5.1.0)
- firebase_core Version: 3.2.0
- Operating System: Windows 10 (Version 10.0.22631.3880)
- CMake Version: 3.30.2

**Error Message:**

```
C:\Users\James\Programming\Flutter\FirstFlutterApp\frequency_details_app\frequency_details_app\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2295,23): error C2146: syntax error: missing ';' before identifier 'encodable_request_arg' [C:\Users\James\Programming\Flutter\FirstFlutterApp\frequency_details_app\frequency_details_app\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
C:\Users\James\Programming\Flutter\FirstFlutterApp\frequency_details_app\frequency_details_app\windows\flutter\ephemeral\.plugin_symlinks\cloud_firestore\windows\messages.g.cpp(2297,28): error C2065: 'request_arg': undeclared identifier [C:\Users\James\Programming\Flutter\FirstFlutterApp\frequency_details_app\frequency_details_app\build\windows\x64\plugins\cloud_firestore\cloud_firestore_plugin.vcxproj]
Error: Build process failed.
```

I also receive numerous warnings like:

`firebase_firestore.lib(7ba01613985f32fe50e0c125a0414f54_firebase_firestore.dir_Debug_aggregate_query_main.obj) : warning LNK4099: PDB '' was not found with 'firebase_firestore.lib(7ba01613985f32fe50e0c125a0414f54_firebase_firestore.dir_Debug_aggregate_query_main.obj)' or at ''; linking object as if no debug info [C:\Users\James\Programming\Flutter\FirstFlutterApp\frequency_details_app\frequency_details_app\build\windows\x64\runner\frequency_details_app.vcxproj]`

**Steps Taken:**

- Verified that cmake is correctly installed and updated to version 3.30.2.
- Updated Visual Studio 2022 with all necessary components, including Desktop development with C++ and Universal Windows Platform (UWP) development.
- Cleaned and rebuilt the project using flutter clean and flutter pub get.
- Attempted to use various versions of cloud_firestore, including 3.1.10 and 5.2.0, to resolve dependency conflicts with firebase_core.
- Removed the caret symbol (^) from the cloud_firestore version to ensure exact versioning.

**pubspec.yaml:**

```
name: frequency_details_app
description: A new Flutter project.
publish_to: 'none'
version: 1.0.0+1

environment:
  sdk: '>=3.4.3 <4.0.0'

dependencies:
  flutter:
    sdk: flutter
  firebase_core: 3.2.0
  firebase_auth: ^5.1.2
  in_app_purchase: ^3.2.0
  collection: ^1.18.0
  http_parser: ^4.0.2
  material_color_utilities: ^0.8.0
  meta: ^1.12.0
  provider: ^6.1.2
  cloud_firestore: 5.2.0
  cupertino_icons: ^1.0.6

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_lints: ^4.0.0
  leak_tracker: ^10.0.4
  leak_tracker_flutter_testing: ^3.0.3
  test_api: ^0.7.0
  vm_service: ^14.2.1

flutter:
  uses-material-design: true
  assets:
    - assets/FrequencyDB.json
```
**Questions:**

- Why am I encountering this syntax error and missing identifier issue with the cloud_firestore plugin on Windows?
- How can I resolve these issues to successfully build my Flutter project on the Windows platform?
- Are there specific configurations or workarounds that I need to apply for compatibility with cloud_firestore on Windows?

Any help or guidance would be greatly appreciated.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,7 +3,7 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
-## 2024-08-06
+## 2024-08-06 - [BoM 2.4.1](https://github.com/firebase/flutterfire/blob/main/VERSIONS.md#flutter-bom-241-2024-08-06)
 
 ### Changes
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,51 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 2024-08-06
+
+### Changes
+
+---
+
+Packages with breaking changes:
+
+ - There are no breaking changes in this release.
+
+Packages with other changes:
+
+ - [`cloud_firestore` - `v5.2.1`](#cloud_firestore---v521)
+ - [`cloud_firestore_platform_interface` - `v6.3.1`](#cloud_firestore_platform_interface---v631)
+ - [`firebase_auth` - `v5.1.4`](#firebase_auth---v514)
+ - [`firebase_storage` - `v12.1.3`](#firebase_storage---v1213)
+ - [`cloud_firestore_web` - `v4.1.1`](#cloud_firestore_web---v411)
+ - [`firebase_vertexai` - `v0.2.2+4`](#firebase_vertexai---v0224)
+
+Packages with dependency updates only:
+
+> Packages listed below depend on other packages in this workspace that have had changes. Their versions have been incremented to bump the minimum dependency versions of the packages they depend upon in this project.
+
+ - `cloud_firestore_web` - `v4.1.1`
+ - `firebase_vertexai` - `v0.2.2+4`
+
+---
+
+#### `cloud_firestore` - `v5.2.1`
+
+ - **FIX**: compilation issue on Windows ([#13135](https://github.com/firebase/flutterfire/issues/13135)). ([de8c9e0f](https://github.com/firebase/flutterfire/commit/de8c9e0f2d3117b3614ac8295b041fea7ed3ee7f))
+
+#### `cloud_firestore_platform_interface` - `v6.3.1`
+
+ - **FIX**: compilation issue on Windows ([#13135](https://github.com/firebase/flutterfire/issues/13135)). ([de8c9e0f](https://github.com/firebase/flutterfire/commit/de8c9e0f2d3117b3614ac8295b041fea7ed3ee7f))
+
+#### `firebase_auth` - `v5.1.4`
+
+ - **FIX**(firebase_auth): added supporting rawNonce for OAuth credential on Windows platform ([#13086](https://github.com/firebase/flutterfire/issues/13086)). ([12e87de9](https://github.com/firebase/flutterfire/commit/12e87de93ddc39d41a6a634d7d03766b3e36996a))
+
+#### `firebase_storage` - `v12.1.3`
+
+ - **FIX**(storage,windows): add log to explain that the Storage Emulator is not available on Windows ([#13147](https://github.com/firebase/flutterfire/issues/13147)). ([8d1ea80c](https://github.com/firebase/flutterfire/commit/8d1ea80cf7b007459572405c876e813b43c3b4cf))
+
+
 ## 2024-07-30 - [BoM 2.4.0](https://github.com/firebase/flutterfire/blob/main/VERSIONS.md#flutter-bom-240-2024-07-30)
 
 ### Changes

--- a/VERSIONS.md
+++ b/VERSIONS.md
@@ -4,6 +4,44 @@ This document is listing all the compatible versions of the FlutterFire plugins.
 
 # Versions
 
+## [Flutter BoM 2.4.1 (2024-08-06)](https://github.com/firebase/flutterfire/blob/main/CHANGELOG.md#2024-08-06)
+
+Install this version using FlutterFire CLI
+
+```bash
+flutterfire install 2.4.1
+```
+
+### Included Native Firebase SDK Versions
+| Firebase SDK | Version | Link |
+|--------------|---------|------|
+| Android SDK | 33.1.0 | [Release Notes](https://firebase.google.com/support/release-notes/android) |
+| iOS SDK | 10.29.0 | [Release Notes](https://firebase.google.com/support/release-notes/ios) |
+| Web SDK | 10.11.1 | [Release Notes](https://firebase.google.com/support/release-notes/js) |
+| Windows SDK | 12.0.0 | [Release Notes](https://firebase.google.com/support/release-notes/cpp-relnotes) |
+
+### FlutterFire Plugin Versions
+| Plugin | Version | Dart Version | Flutter Version |
+|--------|---------|--------------|-----------------|
+| [cloud_firestore](https://pub.dev/packages/cloud_firestore/versions/5.2.1) | 5.2.1 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [cloud_functions](https://pub.dev/packages/cloud_functions/versions/5.0.4) | 5.0.4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_analytics](https://pub.dev/packages/firebase_analytics/versions/11.2.1) | 11.2.1 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_app_check](https://pub.dev/packages/firebase_app_check/versions/0.3.0+4) | 0.3.0+4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_app_installations](https://pub.dev/packages/firebase_app_installations/versions/0.3.0+4) | 0.3.0+4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_auth](https://pub.dev/packages/firebase_auth/versions/5.1.4) | 5.1.4 | >=3.2.0 <4.0.0 | >=3.16.0 |
+| [firebase_core](https://pub.dev/packages/firebase_core/versions/3.3.0) | 3.3.0 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_crashlytics](https://pub.dev/packages/firebase_crashlytics/versions/4.0.4) | 4.0.4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_database](https://pub.dev/packages/firebase_database/versions/11.0.4) | 11.0.4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_dynamic_links](https://pub.dev/packages/firebase_dynamic_links/versions/6.0.4) | 6.0.4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_in_app_messaging](https://pub.dev/packages/firebase_in_app_messaging/versions/0.8.0+4) | 0.8.0+4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_messaging](https://pub.dev/packages/firebase_messaging/versions/15.0.4) | 15.0.4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_ml_model_downloader](https://pub.dev/packages/firebase_ml_model_downloader/versions/0.3.0+4) | 0.3.0+4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_performance](https://pub.dev/packages/firebase_performance/versions/0.10.0+4) | 0.10.0+4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_remote_config](https://pub.dev/packages/firebase_remote_config/versions/5.0.4) | 5.0.4 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_storage](https://pub.dev/packages/firebase_storage/versions/12.1.3) | 12.1.3 | >=3.2.0 <4.0.0 | >=3.3.0 |
+| [firebase_vertexai](https://pub.dev/packages/firebase_vertexai/versions/0.2.2+4) | 0.2.2+4 | >=3.2.0 <4.0.0 | >=3.16.0 |
+
+
 ## [Flutter BoM 2.4.0 (2024-07-30)](https://github.com/firebase/flutterfire/blob/main/CHANGELOG.md#2024-07-30)
 
 Install this version using FlutterFire CLI

--- a/packages/cloud_firestore/cloud_firestore/CHANGELOG.md
+++ b/packages/cloud_firestore/cloud_firestore/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 5.2.1
+
+ - **FIX**: compilation issue on Windows ([#13135](https://github.com/firebase/flutterfire/issues/13135)). ([de8c9e0f](https://github.com/firebase/flutterfire/commit/de8c9e0f2d3117b3614ac8295b041fea7ed3ee7f))
+
 ## 5.2.0
 
  - **FIX**(firestore,web): stop cleaning up snapshot listeners in debug ([#13119](https://github.com/firebase/flutterfire/issues/13119)). ([82a63c8b](https://github.com/firebase/flutterfire/commit/82a63c8bf9bad0c262ed48d7829fb05110a9fe08))

--- a/packages/cloud_firestore/cloud_firestore/example/pubspec.yaml
+++ b/packages/cloud_firestore/cloud_firestore/example/pubspec.yaml
@@ -5,7 +5,7 @@ environment:
   sdk: '>=3.2.0 <4.0.0'
 
 dependencies:
-  cloud_firestore: ^5.2.0
+  cloud_firestore: ^5.2.1
   firebase_core: ^3.3.0
   flutter:
     sdk: flutter

--- a/packages/cloud_firestore/cloud_firestore/pubspec.yaml
+++ b/packages/cloud_firestore/cloud_firestore/pubspec.yaml
@@ -4,7 +4,7 @@ description:
   live synchronization and offline support on Android and iOS.
 homepage: https://firebase.google.com/docs/firestore
 repository: https://github.com/firebase/flutterfire/tree/main/packages/cloud_firestore/cloud_firestore
-version: 5.2.0
+version: 5.2.1
 topics:
   - firebase
   - firestore
@@ -20,8 +20,8 @@ environment:
   flutter: '>=3.3.0'
 
 dependencies:
-  cloud_firestore_platform_interface: ^6.3.0
-  cloud_firestore_web: ^4.1.0
+  cloud_firestore_platform_interface: ^6.3.1
+  cloud_firestore_web: ^4.1.1
   collection: ^1.0.0
   firebase_core: ^3.3.0
   firebase_core_platform_interface: ^5.2.0

--- a/packages/cloud_firestore/cloud_firestore_platform_interface/CHANGELOG.md
+++ b/packages/cloud_firestore/cloud_firestore_platform_interface/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 6.3.1
+
+ - **FIX**: compilation issue on Windows ([#13135](https://github.com/firebase/flutterfire/issues/13135)). ([de8c9e0f](https://github.com/firebase/flutterfire/commit/de8c9e0f2d3117b3614ac8295b041fea7ed3ee7f))
+
 ## 6.3.0
 
  - **FEAT**(firestore): support for `PersistentCacheIndexManager` for firestore instances for managing cache indexes. ([#13070](https://github.com/firebase/flutterfire/issues/13070)). ([806c15d7](https://github.com/firebase/flutterfire/commit/806c15d7eadaf48b8dfb22586bea4ed684672a86))

--- a/packages/cloud_firestore/cloud_firestore_platform_interface/pubspec.yaml
+++ b/packages/cloud_firestore/cloud_firestore_platform_interface/pubspec.yaml
@@ -1,6 +1,6 @@
 name: cloud_firestore_platform_interface
 description: A common platform interface for the cloud_firestore plugin.
-version: 6.3.0
+version: 6.3.1
 homepage: https://github.com/firebase/flutterfire/tree/main/packages/cloud_firestore/cloud_firestore_platform_interface
 repository: https://github.com/firebase/flutterfire/tree/main/packages/cloud_firestore/cloud_firestore_platform_interface
 

--- a/packages/cloud_firestore/cloud_firestore_web/CHANGELOG.md
+++ b/packages/cloud_firestore/cloud_firestore_web/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 4.1.1
+
+ - Update a dependency to the latest release.
+
 ## 4.1.0
 
  - **FIX**(firestore,web): stop cleaning up snapshot listeners in debug ([#13119](https://github.com/firebase/flutterfire/issues/13119)). ([82a63c8b](https://github.com/firebase/flutterfire/commit/82a63c8bf9bad0c262ed48d7829fb05110a9fe08))

--- a/packages/cloud_firestore/cloud_firestore_web/pubspec.yaml
+++ b/packages/cloud_firestore/cloud_firestore_web/pubspec.yaml
@@ -3,7 +3,7 @@ description: The web implementation of cloud_firestore
 homepage: https://github.com/firebase/flutterfire/tree/main/packages/cloud_firestore/cloud_firestore_web
 repository: https://github.com/firebase/flutterfire/tree/main/packages/cloud_firestore/cloud_firestore_web
 
-version: 4.1.0
+version: 4.1.1
 
 environment:
   sdk: '>=3.2.0 <4.0.0'
@@ -11,7 +11,7 @@ environment:
 
 dependencies:
   _flutterfire_internals: ^1.3.40
-  cloud_firestore_platform_interface: ^6.3.0
+  cloud_firestore_platform_interface: ^6.3.1
   collection: ^1.0.0
   firebase_core: ^3.3.0
   firebase_core_web: ^2.17.4

--- a/packages/firebase_app_check/firebase_app_check/example/pubspec.yaml
+++ b/packages/firebase_app_check/firebase_app_check/example/pubspec.yaml
@@ -9,7 +9,7 @@ environment:
   sdk: '>=3.2.0 <4.0.0'
 
 dependencies:
-  cloud_firestore: ^5.2.0
+  cloud_firestore: ^5.2.1
   firebase_app_check: ^0.3.0+4
   firebase_core: ^3.3.0
   flutter:

--- a/packages/firebase_auth/firebase_auth/CHANGELOG.md
+++ b/packages/firebase_auth/firebase_auth/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 5.1.4
+
+ - **FIX**(firebase_auth): added supporting rawNonce for OAuth credential on Windows platform ([#13086](https://github.com/firebase/flutterfire/issues/13086)). ([12e87de9](https://github.com/firebase/flutterfire/commit/12e87de93ddc39d41a6a634d7d03766b3e36996a))
+
 ## 5.1.3
 
  - **DOCS**(auth): add information about error codes for email/password functions ([#13100](https://github.com/firebase/flutterfire/issues/13100)). ([aeafc356](https://github.com/firebase/flutterfire/commit/aeafc356953a0531003f765e766ffcff2387401d))

--- a/packages/firebase_auth/firebase_auth/example/pubspec.yaml
+++ b/packages/firebase_auth/firebase_auth/example/pubspec.yaml
@@ -6,7 +6,7 @@ environment:
 
 dependencies:
   barcode_widget: ^2.0.4
-  firebase_auth: ^5.1.3
+  firebase_auth: ^5.1.4
   firebase_core: ^3.3.0
   firebase_messaging: ^15.0.4
   flutter:

--- a/packages/firebase_auth/firebase_auth/pubspec.yaml
+++ b/packages/firebase_auth/firebase_auth/pubspec.yaml
@@ -4,7 +4,7 @@ description: Flutter plugin for Firebase Auth, enabling
   like Google, Facebook and Twitter.
 homepage: https://firebase.google.com/docs/auth
 repository: https://github.com/firebase/flutterfire/tree/main/packages/firebase_auth/firebase_auth
-version: 5.1.3
+version: 5.1.4
 topics:
   - firebase
   - authentication

--- a/packages/firebase_storage/firebase_storage/CHANGELOG.md
+++ b/packages/firebase_storage/firebase_storage/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 12.1.3
+
+ - **FIX**(storage,windows): add log to explain that the Storage Emulator is not available on Windows ([#13147](https://github.com/firebase/flutterfire/issues/13147)). ([8d1ea80c](https://github.com/firebase/flutterfire/commit/8d1ea80cf7b007459572405c876e813b43c3b4cf))
+
 ## 12.1.2
 
  - Update a dependency to the latest release.

--- a/packages/firebase_storage/firebase_storage/example/pubspec.yaml
+++ b/packages/firebase_storage/firebase_storage/example/pubspec.yaml
@@ -6,7 +6,7 @@ environment:
 
 dependencies:
   firebase_core: ^3.3.0
-  firebase_storage: ^12.1.2
+  firebase_storage: ^12.1.3
   flutter:
     sdk: flutter
   image_picker: ^1.0.2

--- a/packages/firebase_storage/firebase_storage/pubspec.yaml
+++ b/packages/firebase_storage/firebase_storage/pubspec.yaml
@@ -3,7 +3,7 @@ description: Flutter plugin for Firebase Cloud Storage, a powerful, simple, and
   cost-effective object storage service for Android and iOS.
 homepage: https://firebase.google.com/docs/storage/flutter/start
 repository: https://github.com/firebase/flutterfire/tree/main/packages/firebase_storage/firebase_storage
-version: 12.1.2
+version: 12.1.3
 topics:
   - firebase
   - storage

--- a/packages/firebase_vertexai/firebase_vertexai/CHANGELOG.md
+++ b/packages/firebase_vertexai/firebase_vertexai/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 0.2.2+4
+
+ - Update a dependency to the latest release.
+
 ## 0.2.2+3
 
  - Update a dependency to the latest release.

--- a/packages/firebase_vertexai/firebase_vertexai/example/pubspec.yaml
+++ b/packages/firebase_vertexai/firebase_vertexai/example/pubspec.yaml
@@ -20,7 +20,7 @@ dependencies:
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.6
   firebase_core: ^3.3.0
-  firebase_vertexai: ^0.2.2+3
+  firebase_vertexai: ^0.2.2+4
   flutter:
     sdk: flutter
   flutter_markdown: ^0.6.20

--- a/packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_version.dart
+++ b/packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_version.dart
@@ -13,4 +13,4 @@
 // limitations under the License.
 
 /// generated version number for the package, do not manually edit
-const packageVersion = '0.2.2+3';
+const packageVersion = '0.2.2+4';

--- a/packages/firebase_vertexai/firebase_vertexai/pubspec.yaml
+++ b/packages/firebase_vertexai/firebase_vertexai/pubspec.yaml
@@ -1,6 +1,6 @@
 name: firebase_vertexai
 description: "Firebase Vertex AI SDK."
-version: 0.2.2+3
+version: 0.2.2+4
 homepage: https://firebase.google.com/docs/vertex-ai/get-started?platform=flutter
 topics:
   - firebase
@@ -14,7 +14,7 @@ environment:
 
 dependencies:
   firebase_app_check: ^0.3.0+4
-  firebase_auth: ^5.1.3
+  firebase_auth: ^5.1.4
   firebase_core: ^3.3.0
   firebase_core_platform_interface: ^5.2.0
   flutter:

--- a/scripts/versions.json
+++ b/scripts/versions.json
@@ -1,4 +1,32 @@
 {
+  "2.4.1": {
+    "date": "2024-08-06",
+    "firebase_sdk": {
+      "android": "33.1.0",
+      "ios": "10.29.0",
+      "web": "10.11.1",
+      "windows": "12.0.0"
+    },
+    "packages": {
+      "cloud_firestore": "5.2.1",
+      "cloud_functions": "5.0.4",
+      "firebase_analytics": "11.2.1",
+      "firebase_app_check": "0.3.0+4",
+      "firebase_app_installations": "0.3.0+4",
+      "firebase_auth": "5.1.4",
+      "firebase_core": "3.3.0",
+      "firebase_crashlytics": "4.0.4",
+      "firebase_database": "11.0.4",
+      "firebase_dynamic_links": "6.0.4",
+      "firebase_in_app_messaging": "0.8.0+4",
+      "firebase_messaging": "15.0.4",
+      "firebase_ml_model_downloader": "0.3.0+4",
+      "firebase_performance": "0.10.0+4",
+      "firebase_remote_config": "5.0.4",
+      "firebase_storage": "12.1.3",
+      "firebase_vertexai": "0.2.2+4"
+    }
+  },
   "2.4.0": {
     "date": "2024-07-30",
     "firebase_sdk": {

--- a/tests/pubspec.yaml
+++ b/tests/pubspec.yaml
@@ -21,7 +21,7 @@ dependencies:
   firebase_app_installations: ^0.3.0+4
   firebase_app_installations_platform_interface: ^0.1.4+40
   firebase_app_installations_web: ^0.1.5+12
-  firebase_auth: ^5.1.3
+  firebase_auth: ^5.1.4
   firebase_auth_platform_interface: ^7.4.3
   firebase_auth_web: ^5.12.5
   firebase_core: ^3.3.0
@@ -43,7 +43,7 @@ dependencies:
   firebase_remote_config: ^5.0.4
   firebase_remote_config_platform_interface: ^1.4.40
   firebase_remote_config_web: ^1.6.12
-  firebase_storage: ^12.1.2
+  firebase_storage: ^12.1.3
   firebase_storage_platform_interface: ^5.1.27
   firebase_storage_web: ^3.9.12
   flutter:

